src/sources/fluent/mod.rs

```diff
@@ -92,8 +102,50 @@ impl SourceConfig for FluentConfig {
         )
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(DataType::Log)]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        // `host_key` is only inserted if not present already.
+        let host_key = parse_value_path(log_schema().host_key())
+            .ok()
+            .map(LegacyKey::InsertIfEmpty);
+
+        let tag_key = parse_value_path("tag").ok().map(LegacyKey::Overwrite);
+
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        // There is a global and per-source `log_namespace` config.
+        // The source config overrides the global setting and is merged here.
+        let mut schema_definition = BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                FluentConfig::NAME,
+                host_key,
+                &owned_value_path!("host"),
+                Kind::bytes(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                FluentConfig::NAME,
+                tag_key,
+                &owned_value_path!("tag"),
+                Kind::bytes(),
+                Some("tag"),
+            )
+            // for metadata that is added to the events dynamically from the FluentRecord
+            .with_source_metadata(
+                FluentConfig::NAME,
+                None,
+                &owned_value_path!("record"),
+                Kind::object(Collection::empty().with_unknown(Kind::bytes())).or_undefined(),
+                None,
+            );
```

Comment:
Oops, just noticed this is missing the new `fluent.timestamp` metadata value that gets added in the Vector namespace!

Reply:
hummm.. are `amqp` and `aws_sqs` also missing that me thinks?

Reply:
Added to all 3 places in https://github.com/vectordotdev/vector/pull/15206/commits/d1b11ad058d9e44d70df9b740a2bbbc045b97c14